android/src/app/organicmaps/car/screens/NavigationScreen.java

```diff
@@ -55,29 +83,95 @@ public void onPause(@NonNull LifecycleOwner owner)
   public void onNavigationStarted()
   {
     ThemeSwitcher.INSTANCE.restart(true);
+    mNavigationManager.navigationStarted();
   }
 
   @Override
   public void onNavigationCancelled()
   {
     ThemeSwitcher.INSTANCE.restart(true);
+    mNavigationManager.navigationEnded();
+    RoutingUtils.resetTrip();
+    getScreenManager().popToRoot();
+  }
+
+  public void stopNavigation()
+  {
+    mRoutingController.cancel();
   }
 
   @NonNull
   private ActionStrip createActionStrip()
   {
     final Action.Builder stopActionBuilder = new Action.Builder();
     stopActionBuilder.setTitle(getCarContext().getString(R.string.current_location_unknown_stop_button));
-    stopActionBuilder.setOnClickListener(this::stop);
+    stopActionBuilder.setOnClickListener(this::stopNavigation);
+
     final ActionStrip.Builder builder = new ActionStrip.Builder();
-    builder.addAction(UiHelpers.createSettingsAction(this, getSurfaceRenderer()));
+    builder.addAction(createTtsAction());
+    builder.addAction(UiHelpers.createSettingsActionForResult(this, getSurfaceRenderer(), this::onSettingsResult));
     builder.addAction(stopActionBuilder.build());
     return builder.build();
   }
 
-  private void stop()
+  @Nullable
+  private TravelEstimate getDestinationTravelEstimate()
   {
-    mRoutingController.cancel();
-    getScreenManager().popToRoot();
+    final Trip trip = RoutingUtils.getLastTrip();
+
+    if (trip.isLoading())
+      return null;
+
+    List<TravelEstimate> travelEstimates = trip.getDestinationTravelEstimates();
+    if (travelEstimates.size() != 1)
+      throw new RuntimeException("TravelEstimates size must be 1");
```

Comment:
It must be 1 for our case. 
It is possible to specify distance and time estimates to every point in route (intermediate + finish). That's why array is returned.

Do we have such functionality (to get travel estimate for each point) in core?
It will be good support it here.